src/workspaces/features/GeneratePipelineWebhookUrlDialog/GeneratePipelineWebhookUrlDialog.tsx

```diff
@@ -0,0 +1,98 @@
+import Button from "core/components/Button";
+import Dialog from "core/components/Dialog";
+import Spinner from "core/components/Spinner";
+import { useState } from "react";
+import { Trans, useTranslation } from "next-i18next";
+import { gql } from "@apollo/client";
+import { GeneratePipelineWebhookUrlError } from "graphql/types";
+import { AlertType, displayAlert } from "core/helpers/alert";
+import { ExclamationCircleIcon } from "@heroicons/react/24/outline";
+import useCacheKey from "core/hooks/useCacheKey";
+import { GeneratePipelineWebhookUrlDialog_PipelineFragment } from "./GeneratePipelineWebhookUrlDialog.generated";
+import { useGenerateWebhookPipelineWebhookUrlMutation } from "workspaces/graphql/mutations.generated";
+
+type GeneratePipelineWebhookUrlDialogProps = {
+  onClose(): void;
+  open: boolean;
+  pipeline: GeneratePipelineWebhookUrlDialog_PipelineFragment;
+};
+
+const GeneratePipelineWebhookUrlDialog = (
+  props: GeneratePipelineWebhookUrlDialogProps,
+) => {
+  const { open, onClose, pipeline } = props;
+  const { t } = useTranslation();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const clearCache = useCacheKey(["pipelines", pipeline.code]);
+
+  const [generateWebhookUrl] = useGenerateWebhookPipelineWebhookUrlMutation();
+  const onSubmit = async () => {
+    setIsSubmitting(true);
+    const { data } = await generateWebhookUrl({
+      variables: {
+        input: {
+          id: pipeline.id,
+        },
+      },
+    });
+
+    if (!data?.generatePipelineWebhookUrl) {
+      throw new Error("Unknown error.");
+    }
+
+    if (data.generatePipelineWebhookUrl.success) {
+      displayAlert(t("New url successfully created"), AlertType.info);
+      onClose();
+      clearCache();
```

Comment:
There is no need to do that if you get back the pipeline with the webhookToken and its identifier.